colossalai/tensor/shape_consistency.py

```diff
@@ -0,0 +1,306 @@
+import torch
+from colossalai.tensor.sharding_spec import ShardingSpec, _DimSpec
+from enum import Enum
+from copy import deepcopy
+
+
+class CollectiveCommPattern(Enum):
+    ALLGATHER = 'all_gather'
+    ALLTOALL = 'all_to_all'
+    SHARD = 'shard'
```

Comment:
This enum is not used anywhere in the code?

Reply:
It will be used in apply cached_spec_dict, but this part is WIP.